src/Agent.Plugins/BuildArtifact/BuildArtifactPluginV1.cs

```diff
@@ -0,0 +1,382 @@
+
+            var downloadOption = downloadType == "single" ? DownloadOptions.SingleDownload : DownloadOptions.MultiDownload;
+
+            context.Output(StringUtil.Loc("DownloadArtifactTo", targetPath));
+            await server.DownloadAsyncV2(context, downloadParameters, downloadOption, token);
+            context.Output(StringUtil.Loc("DownloadArtifactFinished"));
+        }
+
+        private string CreateDirectoryIfDoesntExist(string targetPath)
+        {
+            string fullPath = Path.GetFullPath(targetPath);
+            bool dirExists = Directory.Exists(fullPath);
+            if (!dirExists)
+            {
+                Directory.CreateDirectory(fullPath);
+            }
+            return fullPath;
+        }
+
+        private async Task<int> GetPipelineIdAsync(AgentTaskPluginExecutionContext context, string pipelineDefinition, string buildVersionToDownload, string project, string[] tagFilters, BuildResult resultFilter = BuildResult.Succeeded, string branchName = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if(String.IsNullOrWhiteSpace(pipelineDefinition)) 
+            {
+                throw new InvalidOperationException(StringUtil.Loc("CannotBeNullOrEmpty", "Pipeline Definition"));
+            }
+
+            VssConnection connection = context.VssConnection;
+            BuildHttpClient buildHttpClient = connection.GetClient<BuildHttpClient>();
+
+            var isDefinitionNum = Int32.TryParse(pipelineDefinition, out int definition);
+            if (!isDefinitionNum)
+            {
+                var definitionRef = (await buildHttpClient.GetDefinitionsAsync(new System.Guid(project), pipelineDefinition, cancellationToken: cancellationToken)).FirstOrDefault();
+                if (definitionRef == null)
+                {
+                    throw new ArgumentException(StringUtil.Loc("PipelineDoesNotExist", pipelineDefinition));
+                }
+                else
+                {
+                    definition = definitionRef.Id;
+                }
+            }
+            var definitions = new List<int>() { definition };
+
+            List<Build> list;
+            if (buildVersionToDownload == buildVersionToDownloadLatest)
+            {
+                list = await buildHttpClient.GetBuildsAsync(project, definitions, tagFilters: tagFilters, queryOrder: BuildQueryOrder.FinishTimeDescending, resultFilter: resultFilter);
+            }
+            else if (buildVersionToDownload == buildVersionToDownloadLatestFromBranch)
+            {
+                list = await buildHttpClient.GetBuildsAsync(project, definitions, branchName: branchName, tagFilters: tagFilters, queryOrder: BuildQueryOrder.FinishTimeDescending, resultFilter: resultFilter);
+            }
+            else
+            {
+                throw new InvalidOperationException("Unreachable code!");
+            }
+
+            if (list.Count > 0)
```

Comment:
> 0 or == 1?

Reply:
I think this is correct because we want the latest if there are more than one